src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -135,25 +142,37 @@ public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
             while (sourceIndex < srcLength - 2)
             {
                 result = Encode(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex > destLength - 4) goto FalseExit;
```

Comment:
Surprisingly, this change severely degrades perf:
![image](https://cloud.githubusercontent.com/assets/6527137/26387965/800790a4-4006-11e7-89e3-6887f33fbd59.png)

I think it maybe due to swapping variables in/out of registers.

The way it is now, we use sourceIndex in the while loop check and then right after to encode and similarly we use destIndex for the buffer space check and right after to write.
If we swap the order, we lose the re-use of variables.